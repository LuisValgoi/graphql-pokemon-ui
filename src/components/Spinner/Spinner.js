import React from 'react';

import Spinner from 'react-bootstrap/Spinner';

function SpinnerInternal() {
  return <Spinner style={{ margin: '0 auto' }} animation="border" />;
}

export default SpinnerInternal;
