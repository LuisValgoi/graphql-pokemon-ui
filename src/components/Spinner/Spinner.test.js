import React from 'react';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Spinner from './Spinner';

afterEach(cleanup);

test('should render', () => {
  const { asFragment } = render(<Spinner />);
  expect(asFragment()).toMatchSnapshot();
})

test('should has text-center class', () => {
  const { getByTestId } = render(<Spinner />);
  expect(getByTestId('spinner')).toHaveClass('text-center');
})
