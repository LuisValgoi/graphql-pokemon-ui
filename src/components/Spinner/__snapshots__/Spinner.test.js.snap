// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<DocumentFragment>
  <div
    class="text-center"
    data-testid="spinner"
    style="margin-top: 1rem;"
  >
    <div
      class="spinner-border"
    />
  </div>
</DocumentFragment>
`;

exports[`spinner should render 1`] = `
<DocumentFragment>
  <div
    class="text-center"
    data-testid="spinner"
    style="margin-top: 1rem;"
  >
    <div
      class="spinner-border"
    />
  </div>
</DocumentFragment>
`;
